test/e2e/framework/context.go

```diff
@@ -0,0 +1,59 @@
+package framework
+
+import (
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type TestCtx struct {
+	ID         string
+	cleanUpFns []finalizerFn
+}
+
+type finalizerFn func() error
+
+func (f *Framework) NewTestCtx(t *testing.T) TestCtx {
+	// TestCtx is used among others for namespace names where '/' is forbidden
+	prefix := strings.TrimPrefix(
+		strings.Replace(
+			strings.ToLower(t.Name()),
+			"/",
+			"-",
+			-1,
+		),
+		"test",
+	)
+
+	id := prefix + "-" + strconv.FormatInt(time.Now().Unix(), 10)
+	return TestCtx{
+		ID: id,
+	}
+}
+
+// GetObjID returns an ascending ID based on the length of cleanUpFns. It is
+// based on the premise that every new object also appends a new finalizerFn on
+// cleanUpFns. This can e.g. be used to create multiple namespaces in the same
+// test context.
+func (ctx *TestCtx) GetObjID() string {
+	return ctx.ID + "-" + strconv.Itoa(len(ctx.cleanUpFns))
+}
+
+func (ctx *TestCtx) Cleanup(t *testing.T) {
+	var eg errgroup.Group
+
+	for i := len(ctx.cleanUpFns) - 1; i >= 0; i-- {
+		eg.Go(ctx.cleanUpFns[i])
```

Comment:
We should not run the cleanup functions concurrently in case they conflict with each other in the event of a race condition.